tests/test_on_poll.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/python
+from collections.abc import Iterator
+import pytest
+from unittest import mock
+
+from soar_sdk.app import App
+from soar_sdk.params import OnPollParams
+
+
+def test_on_poll_decoration_fails_when_used_more_than_once(simple_app: App):
+    """Test that the on_poll decorator can only be used once per app."""
+
+    @simple_app.on_poll()
+    def on_poll_function(params: OnPollParams) -> Iterator[dict]:
+        yield {"data": "test"}
+
+    with pytest.raises(TypeError) as exception_info:
+
+        @simple_app.on_poll()
+        def second_on_poll(params: OnPollParams) -> Iterator[dict]:
+            yield {"data": "another test"}
+
+    assert "The 'on_poll' decorator can only be used once per App instance." in str(
+        exception_info
+    )
```

Comment:
Nit: prefer `match`. Also, I usually try and make it possible to change the error message slightly without breaking tests, by only searching for keywords in the error:

```suggestion
    with pytest.raises(TypeError, match=r'on_poll.+once per'):

        @simple_app.on_poll()
        def second_on_poll(params: OnPollParams) -> Iterator[dict]:
            yield {"data": "another test"}

```

Reply:
Updated 👍